docs/_scripts/notebook_hooks.py

```diff
@@ -57,7 +64,25 @@ def on_files(files: Files, **kwargs: Dict[str, Any]):
     return new_files
 
 
-def _highlight_code_blocks(markdown: str) -> str:
+def _add_path_to_code_blocks(markdown: str, page: Page) -> str:
+    """Add the path to the code blocks."""
+    code_block_pattern = re.compile(
+        r"(?P<indent>[ \t]*)```(?P<language>\w+)[ ]*(?P<attributes>[^\n]*)\n"
```

Comment:
We should filter this on being exec={on} since not all code blocks are executable at the moment (e.g., some conceptual pages have scaffolds)